### PR TITLE
Fix multiple triggers of the same type adding to DI

### PR DESCRIPTION
**Pull request by alrod**
`serviceProvider.GetRequiredService<T>` returns first service `T` in the service provider. As we have multiple `T` services we need to filter by function Id

Related to the recent commit:
https://github.com/Azure/azure-sdk-for-net/commit/7f267723e231e064015c5f98cb9e332dbbabc584

### CODE REVIEW

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusHostBuilderExtensions.cs [hunk 1]
```diff
@@ -109,8 +110,8 @@ public static IWebJobsBuilder AddServiceBus(this IWebJobsBuilder builder, Action
         internal static IWebJobsBuilder AddServiceBusScaleForTrigger(this IWebJobsBuilder builder, TriggerMetadata triggerMetadata)
         {
             builder.Services.AddSingleton(serviceProvider => new ServiceBusScalerProvider(serviceProvider, triggerMetadata));
-            builder.Services.AddSingleton<IScaleMonitorProvider>(serviceProvider => serviceProvider.GetRequiredService<ServiceBusScalerProvider>());
-            builder.Services.AddSingleton<ITargetScalerProvider>(serviceProvider => serviceProvider.GetRequiredService<ServiceBusScalerProvider>());
+            builder.Services.AddSingleton<IScaleMonitorProvider>(serviceProvider => serviceProvider.GetServices<ServiceBusScalerProvider>().Where(x => x.GetMonitor().Descriptor.FunctionId == triggerMetadata.FunctionName).First());
```
Comment (JoshLove-msft) [inline sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusHostBuilderExtensions.cs:113]: Why do we need to add the two interface singletons? I don't see us injecting this interface anywhere and we've already added the ServiceBusScalerProvider on L112.

Comment (alrod) [reply sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusHostBuilderExtensions.cs:113]: `ServiceBusScalerProvider` implements both `IScaleMonitorProvider` and `ITargetScalerProvider` and we want to create one instance of `ServiceBusScalerProvider`. Web Jobs SDK needs to resolve both `IScaleMonitorProvider` and `ITargetScalerProvider` so we are adding mapping to the `ServiceBusScalerProvider` instance added.

Comment (JoshLove-msft) [reply sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusHostBuilderExtensions.cs]: Thanks for that context. Can you explain why we would have more than a single instance of ServiceBusScalerProvider? Isn't there only a single instance of the SB extension in the running process?

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/EventHubWebJobsBuilderExtensions.cs [hunk 2]
```diff
@@ -94,9 +95,14 @@ public static IWebJobsBuilder AddEventHubs(this IWebJobsBuilder builder, Action<
 
         internal static IWebJobsBuilder AddEventHubsScaleForTrigger(this IWebJobsBuilder builder, TriggerMetadata triggerMetadata)
         {
-            builder.Services.AddSingleton(serviceProvider => new EventHubsScalerProvider(serviceProvider, triggerMetadata));
-            builder.Services.AddSingleton<IScaleMonitorProvider>(serviceProvider => serviceProvider.GetRequiredService<EventHubsScalerProvider>());
-            builder.Services.AddSingleton<ITargetScalerProvider>(serviceProvider => serviceProvider.GetRequiredService<EventHubsScalerProvider>());
+            EventHubsScalerProvider serviceBusScalerProvider = null;
```
Comment (JoshLove-msft) [inline sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/EventHubWebJobsBuilderExtensions.cs:98]:
```suggestion
            EventHubsScalerProvider eventHubsScalerProvider = null;
```

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/EventHubWebJobsBuilderExtensions.cs [hunk 3]
```diff
@@ -94,9 +95,14 @@ public static IWebJobsBuilder AddEventHubs(this IWebJobsBuilder builder, Action<
...
+            builder.Services.AddSingleton(serviceProvider =>
+            {
+                serviceBusScalerProvider = new EventHubsScalerProvider(serviceProvider, triggerMetadata);
```
Comment (JoshLove-msft) [inline sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/EventHubWebJobsBuilderExtensions.cs:101]:
```suggestion
                eventHubsScalerProvider = new EventHubsScalerProvider(serviceProvider, triggerMetadata);
```

File: sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/EventHubWebJobsBuilderExtensions.cs [hunk 4]
```diff
@@ -94,9 +95,14 @@ public static IWebJobsBuilder AddEventHubs(this IWebJobsBuilder builder, Action<
...
+                return serviceBusScalerProvider;
```
Comment (JoshLove-msft) [inline sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/EventHubWebJobsBuilderExtensions.cs:102]:
```suggestion
                return eventHubsScalerProvider;
```

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusHostBuilderExtensions.cs (hunk 1 above)
Comment (alrod) [reply sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusHostBuilderExtensions.cs]: We are adding ServiceBusScalerProvider for each trigger (if the app has multiple SB triggers).

**Review by JoshLove-msft — APPROVED** [review]
LGTM, we should be able to merge once the SB and EH test runs are green.

**Pull request merged**